README.md

```diff
@@ -12,36 +12,61 @@
-### Run development server
+### Local Setup
 
-````
-./tools/run.sh
-````
+To configure your development environment on your system, please follow these steps carefully.
+
+1. Ensure that the following packages are installed alongside your preferred IDE:
+   - `npm` version 7 or later
+   - `nodejs` version 18 or later
+   - `python3` version 3.11 or later
+   - `python3-pip` (Debian-based distributions) / `python-pip` (Arch-based distributions)
+   - `python3-venv` (only on Debian-based distributions)
+   - `gettext` for translation features
+   - Either `postgresql` **or** `docker` to run a local database server
+2. Execute `tools/install.sh` to download all dependencies.
+3. Execute `tools/migrate.sh` to apply all database schema migrations.
+4. Optionally, run `tools/loadtestdata.sh` to apply a predefined set of test data.
+
+### Devcontainer
+
+To configure your development container, please follow these steps carefully.
+
+1. Make sure you have Docker and VSCode installed on your machine.
+2. Open the project in VSCode.
+3. If you're opening VSCode for the first time, you'll be prompted to install the "Remote - Containers" extension. Click "Install" to proceed.
+4. Open the command palette (Ctrl + Shift + P or Cmd + Shift + P on macOS) and search for "Remote-Containers: Open Folder in Container".
+5. Select the directory of your project.
+6. VSCode will open the project in a new container, install all further required tools and load the testdata.
+
+#### Known Limitations
+
+The perfect is the enemy of the good; thus, this section illuminates aspects of our evolving development setup.
+
+##### Sharing git username and email
+
+A known limitation exists where certain versions of Visual Studio Code (VSCode) may not copy the user's `.gitconfig` file correctly into the Devcontainer environment. In such cases, when you attempt to commit changes within the Devcontainer, you may be prompted to enter your Git username and email every time. This can be inconvenient and disrupt the workflow. However, there is a workaround for this issue. You can resolve it by appending the content of your personal `.gitconfig` file, located at `$HOME/.gitconfig`, to the end of the repository-specific `.git/config` file, which in this case would be `integreat-cms/.git/config`. By doing so, you ensure that the necessary Git configuration settings are correctly applied within the Devcontainer environment, allowing for a smoother development experience.
+
+For more information, refer to [this issue](https://github.com/microsoft/vscode-remote-release/issues/1729) in the VSCode Remote Extension repository.
+
+##### Test discovery 
+
+Testing is not yet fully integrated into our Visual Studio Code Container, but we provide convenient task definitions to facilitate testing. For example, you can simply type task current python test and hit enter to run the current Python test file. Take a look at the `.vscode/tasks.json` for more helpful tasks and extend this file as needed.
```

Comment:
```suggestion
Testing is not yet fully integrated into our Visual Studio Code Container, but we provide convenient task definitions to facilitate testing. For example, you can simply type `task current python test` and hit enter to run the current Python test file. Take a look at the `.vscode/tasks.json` for more helpful tasks and extend this file as needed.
```
(I assume this is what was meant? TBH I don't quite get this sentence, but then again I'm not really vamiliar with VSCode)

Reply:
You've got it right! It's possible to run tests within vscode using an extension that adds a small run button next to each test definition, but I couldn't make it work in our setup. However, the loom I linked in my other comment demonstrates the alternative path using tasks, which makes it a bit more convenient to run the scripts. It's not perfect, but it's better than before. :D